plugins/woocommerce/src/Admin/API/AnalyticsImports.php

```diff
@@ -0,0 +1,279 @@
+	 *
+	 * @param  \WP_REST_Request $request Full details about the request.
+	 * @return \WP_Error|boolean
+	 */
+	public function get_item_permissions_check( $request ) {
+		if ( ! current_user_can( 'manage_woocommerce' ) ) {
+			return new \WP_Error(
+				'woocommerce_rest_cannot_view',
+				__( 'Sorry, you cannot view analytics import status.', 'woocommerce' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check if a given request has access to trigger import.
+	 *
+	 * @param  \WP_REST_Request $request Full details about the request.
+	 * @return \WP_Error|boolean
+	 */
+	public function create_item_permissions_check( $request ) {
+		if ( ! current_user_can( 'manage_woocommerce' ) ) {
+			return new \WP_Error(
+				'woocommerce_rest_cannot_create',
+				__( 'Sorry, you cannot trigger analytics imports.', 'woocommerce' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get the current import status.
+	 *
+	 * @param  \WP_REST_Request $request Full details about the request.
+	 * @return \WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_status( $request ) {
+		$is_immediate_mode = $this->is_immediate_import_enabled();
+		$mode              = $is_immediate_mode ? 'immediate' : 'scheduled';
+
+		$response = array(
+			'mode'                              => $mode,
+			'last_processed_date'               => null,
+			'next_scheduled'                    => null,
+			'manual_triggered_import_scheduled' => null,
+		);
+
+		// For scheduled mode, populate additional fields.
+		if ( ! $is_immediate_mode ) {
+			$last_processed_gmt              = get_option( OrdersScheduler::LAST_PROCESSED_ORDER_DATE_OPTION, null );
+			$response['last_processed_date'] = $last_processed_gmt ? get_date_from_gmt( $last_processed_gmt, 'Y-m-d H:i:s' ) : null;
+			$response['next_scheduled']      = $this->get_next_scheduled_time();
+
+			$response['manual_triggered_import_scheduled'] = $this->has_manual_triggered_import_scheduled();
+		}
+
+		return rest_ensure_response( $response );
+	}
+
+	/**
+	 * Trigger a manual import.
+	 *
+	 * @param  \WP_REST_Request $request Full details about the request.
+	 * @return \WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function trigger_import( $request ) {
+		$is_immediate_mode = $this->is_immediate_import_enabled();
+
+		// Return error if in immediate mode.
+		if ( $is_immediate_mode ) {
+			return new \WP_Error(
+				'woocommerce_rest_analytics_import_immediate_mode',
+				__( 'Manual import is not available in immediate mode. Imports happen automatically.', 'woocommerce' ),
+				array( 'status' => 400 )
+			);
+		}
+
+		// Determine if a batch import has already been scheduled or is in progress, excluding the recurring import action.
+		if ( $this->has_manual_triggered_import_scheduled() ) {
```

Comment:
I ended up have two scheduled actions after manually triggering it from the endpoint. I was thinking the check would short circuit the endpoint, but it didn't catch it, it seems:

<img width="774" height="479" alt="Image" src="https://github.com/user-attachments/assets/6c332c35-26e0-435c-8291-4530b7872ff2" />

Reply:
This behavior is actually expected. Manually triggering the action creates one instance, and the recurring schedule creates another. However, this can be a bit confusing. It might be possible to trigger the recurring action immediately instead of scheduling an additional one. I'll investigate this approach to see if it offers a clearer solution.

Reply:
I've updated the PR to ensure that only one action is scheduled at a time. When manually triggering the api call, the recurring action will be canceled first and then rescheduled to run immediately.